Remove redundant line in os_port

dominikholler opened the pull request:
The renaming of the key 'vnic_type' to 'binding:vnic_type'
is reduced to a single line.
The old key is removed to avoid to possibility of
inconsistency.

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

Just a code improvement, no change in behavior.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### COMPONENT NAME
os_port

dominikholler committed:
Remove redundant line in os_port

The renaming of the key 'vnic_type' to 'binding:vnic_type'
is reduced to a single line.
The old key is removed to avoid to possibility of
inconsistency.

dominikholler commented:
@abadger @MaxBab Can you have a look?

gtema commented on lib/ansible/modules/cloud/openstack/os_port.py:
```diff
@@ -356,8 +356,7 @@ def main():
         # for the port type.
-        module.params['binding:vnic_type'] = module.params['vnic_type']
-        module.params.pop('vnic_type', None)
+        module.params['binding:vnic_type'] = module.params.pop('vnic_type')
```
default is missing to allow skipping this param

dominikholler replied:
> default is missing to allow skipping this param

by intention, because the key should exist. From my point of view, it is a good style to explode, if the key is expected, but missing. But I am open to keeping the default value if you think this would be good here.

gtema replied:
Well, if you think it is mandatory - you need to tell this to ansible (line 329). However Neutron explicitly states it is optional (see https://developer.openstack.org/api-ref/network/v2/index.html?expanded=create-port-detail#create-port). Parameter recently added can not be mandatory not breaking older clients.

dominikholler replied:
Line 329 defines 'None' as the default value, so the key will be in dict, even the value might be none.

MaxBab commented:
Works for me.
Tested on my environment.
LGTM.

Could you update the example of the port creation, please?
I forgot to mention the network argument in the initial change and it is mandatory.
Lines: 156-165

```yaml
# Create port of type 'direct'
- os_port:
    state: present
    auth:
      auth_url: https://identity.example.com
      username: admin
      password: admin
      project_name: admin
    name: port1
    network: foo
    vnic_type: direct
```

dominikholler commented:
On Fri, 14 Jun 2019 11:07:53 -0700
Maxim Babushkin <notifications@github.com> wrote:

> Works for me.
> Tested on my environment.
> LGTM.
>
> Could you update the example of the port creation, please?
> I forgot to mention the network argument in the initial change and it
> is mandatory. Lines: 156-165
>
> ```yaml
> # Create port of type 'direct'
> - os_port:
>     state: present
>     auth:
>       auth_url: https://identity.example.com
>       username: admin
>       password: admin
>       project_name: admin
>     name: port1
>     network: foo
>     vnic_type: direct
> ```
>

good idea, I will update, but may take a little bit of time.

dominikholler committed:
Add missing parameter in example of os_port

In the example "Create port of type 'direct'" the parameter
"network" is added, because this parameter is required on
creating new ports.

dominikholler commented:
> > Could you update the example of the port creation, please?
> > I forgot to mention the network argument in the initial change and it
> > is mandatory. Lines: 156-165
> >
> > ```yaml
> > # Create port of type 'direct'
> > - os_port:
> >     state: present
> >     auth:
> >       auth_url: https://identity.example.com
> >       username: admin
> >       password: admin
> >       project_name: admin
> >     name: port1
> >     network: foo
> >     vnic_type: direct
> > ```
> >
>
> good idea, I will update, but may take a little bit of time.

Done.

MaxBab commented:
Thanks a lot for adding it.

cloudnull reviewed (commented):
:+1: LGTM

cloudnull commented:
shipit!

MaxBab commented:
@Shrews @cloudnull @dagnello @emonty @gtema @juliakreger @mnaser @odyssey4me @rcarrillocruz

Hi guys,
Is there a chance to prioritize it, please?
We really need that fix in the module.

Thank you,
Maxim.

gtema commented:
recheck

gtema commented:
shipit
